src/index.js

```diff
@@ -265,15 +262,9 @@ class FsDatastore {
       delete (key /* : Key */) /* : void */ {
         deletes.push(key)
       },
-      commit: (callback /* : (err: ?Error) => void */) => {
-        waterfall([
-          (cb) => each(puts, (p, cb) => {
-            this.put(p.key, p.value, cb)
-          }, cb),
-          (cb) => each(deletes, (k, cb) => {
-            this.delete(k, cb)
-          }, cb)
-        ], (err) => callback(err))
+      commit: async () /* :  Promise<void> */ => {
+        await Promise.all((puts.map((put) => this.put(put.key, put.value))))
```

Comment:
It may be more efficient to put these in one big `Promise.all()` instead of two separate ones